code/modules/mob_spawn/mob_spawn.dm

```diff
@@ -172,8 +172,11 @@
 	user.log_message("became a [prompt_name].", LOG_GAME)
 	uses -= 1 // Remove a use before trying to spawn to prevent strangeness like the spawner trying to spawn more mobs than it should be able to
 
-	if(!(create(user)))
-		message_admins("[src] didn't return anything when creating a mob, this might be broken! The use of the spawner it would have taken has been refunded.")
+	var/created = create(user)
+	if(!created)
+		if(created != MOB_SPAWN_REJECTED)
+			message_admins("[src] didn't return anything when creating a mob, this might be broken! The use of the spawner it would have taken has been refunded. [ADMIN_FLW(src)]")
+
 		uses += 1 // Oops! We messed up and somehow the mob didn't spawn, but that's alright we can refund the use.
```

Comment:
Yeah we definitely could do a falsey check and an `isnull` check, I just wasn't sure if returning something explicit would be clearer behaviour. If you prefer that I'll swap them round.

Reply:
i mean couldn't it just be

```txt
if(isnull(thing))
    message_admins("message")
    uses++
```

without changing anything else?

Reply:
No, we need to increment `uses` in both cases (because the use is expended before it tries to spawn something) but only print the "something went wrong" message if we didn't explicitly return false.

Reply:
alright. implementation already looks much better than what it was prior so i'm happy